benchmarks/sum-file.lua

```diff
@@ -0,0 +1,6 @@
+
+local sum = 0
+for line in io.lines() do
+  sum = sum + line
```

Comment:
I wasn't expecting that to work! Seems Lua will parse a string looking for a numeric value if you use `+` on it!

```
> print(10 + "11")
21
```

Python:
```
>>> print(10 + "11")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

Today I learned.

Reply:
Lua in very early versions started life as something very close to json as i understand it. Also this benchmark i didn't even convert since it just based on reading from stdio really should be removed

Reply:
If you think it's not useful, then I'm OK with you killing it. Up to you.